sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -675,9 +679,11 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]]
 }
 
 object QueryPlan extends PredicateHelper {
-  val OP_ID_TAG = TreeNodeTag[Int]("operatorId")
   val CODEGEN_ID_TAG = new TreeNodeTag[Int]("wholeStageCodegenId")
 
+  val localIdMap: ThreadLocal[java.util.Map[QueryPlan[_], Int]] = ThreadLocal.withInitial(() =>
```

Comment:
can we define the scope of this thread local? When it's set and when it's cleared.

Reply:
the scope is `ExplainUtils.processPlan`, but I defined it here because `QueryPlan` also need this, and don't have access to `execution` package from `catalyst`.  Added comments to clarify.